fearless/Common/Codable/JSON+Optional.swift

```diff
@@ -0,0 +1,26 @@
+import Foundation
+
+private struct OptionalDecodable<T: Decodable>: Decodable {
+    let base: T?
+
+    init(from decoder: Decoder) throws {
+        let decoder = try decoder.singleValueContainer()
+        base = try? decoder.decode(T.self)
+    }
+}
+
+extension KeyedDecodingContainer {
+    func decodeOptionalArray<T: Decodable>(_: [T].Type, forKey key: K) -> [T] {
```

Comment:
decodeOptionalArray from the name means that you have optional as result. decodeRequiredArray or something like decodeWithUnwrap should be there IMO